enclave/src/openfinex/client_utils.rs

```diff
@@ -0,0 +1,221 @@
+            (u16::from_be_bytes(slice) as u64, buffer[4 ..].to_vec())
+        }
+        _   => (pay_len as u64, buffer[2 ..].to_vec())
+    };
+    debug!("payload_length: {}", payload_length);
+
+    // payloads larger than 125 bytes are not allowed for control frames
+    match opcode {
+        CloseOp | PingOp if payload_length > 125 => panic!(),
+        _ => ()
+    }
+
+    // No mask from server
+    /* let masking_key = try!(stream.read_exact(4));
+    let mut masked_payload_buf = try!(stream.read_exact(payload_length as uint));
+    // unmask the payload in-place
+    for (i, octet) in masked_payload_buf.iter_mut().enumerate() {
+        *octet = *octet ^ masking_key[i % 4];
+    }
+    let payload_buf = masked_payload_buf; */
+
+    let payload: Payload = match opcode {
+        TextOp   => {
+            error!("Message: {}", String::from_utf8(payload_buf.to_vec()).unwrap());
+            Payload::Text(String::from_utf8(payload_buf.to_vec()).unwrap())
+        },
+        BinaryOp => Payload::Binary(payload_buf.to_vec()),
+        CloseOp  => Payload::Empty,
+        PingOp   => {
+            error!("Ping");
+            Payload::Binary(payload_buf.to_vec())
+
+        },
+        PongOp   => {
+            error!("Pong");
+            Payload::Binary(payload_buf.to_vec())
+        },
+        _        => unimplemented!(), // ContinuationOp
+    };
+
+    // for now only take text option
+    return Some(Message::new(payload, opcode))
+}
+
+
+// https://datatracker.ietf.org/doc/html/rfc6455#section-5
+pub fn mask(payload: &[u8], opcode: Opcode) -> Vec<u8> {
+    let mut vec = Vec::<u8>::new();
+    vec.push(0b10000000 | opcode as u8); // fin: 1, rsv: 000
+
+    let masking_bit = 0b10000000;
+    let payload_len = payload.len();
+    if payload_len <= 125 {
+        vec.push((masking_bit | payload_len) as u8);
+    } else if payload_len <= 65535 {
+        vec.push((masking_bit | 0b01111110) as u8);
+        let mut length = (payload_len as u16).to_be_bytes().to_vec();
+        vec.append(&mut length);
+    } else if payload_len as u64 <= u64::MAX {
+        vec.push((masking_bit | 0b01111111) as u8);
+        let mut length = (payload_len as u64).to_be_bytes().to_vec();
+        vec.append(&mut length);
+    } else {
+        error!("too long payload");
+        return vec![]
+    }
+    let mask_key = gen_mask();
+    let mut masked_payload = mask_data(mask_key, payload);
+    vec.append(&mut mask_key.to_vec());
+    debug!("bytes: {:?}", vec);
+    vec.append(&mut masked_payload);
+    debug!("length: {:?}", vec.len());
+    vec
+
+}
+
+
+
+/// Generates a random masking key
+pub fn gen_mask() -> [u8; 4] {
+	sgx_rand::random()
+}
+
+/// Masks data to send to a server and writes
+pub fn mask_data(mask: [u8; 4], data: &[u8]) -> Vec<u8> {
+	let mut out = Vec::with_capacity(data.len());
+	let zip_iter = data.iter().zip(mask.iter().cycle());
+	for (&buf_item, &key_item) in zip_iter {
+		out.push(buf_item ^ key_item);
+	}
+	out
+}
+
```

Comment:
No unit tests for this client, because it is temporary anyway and will be exchanged with the rustls client as soon as openfinex supports tls.